pyramid/config/security.py

```diff
@@ -218,9 +225,25 @@ def register():
         intr['header'] = header
         intr['safe_methods'] = as_sorted_tuple(safe_methods)
         intr['callback'] = callback
+
         self.action(IDefaultCSRFOptions, register, order=PHASE1_CONFIG,
                     introspectables=(intr,))
 
+    @action_method
+    def set_csrf_storage_policy(self, policy):
+        """
+        Set the CSRF storage policy used by subsequent view registrations.
+
+        ``policy`` is a class that implements the
+        :meth:`pyramid.interfaces.ICSRFStoragePolicy` interface that will be used for all
+        CSRF functionality.
+        """
+        def register():
+            self.registry.registerUtility(policy, ICSRFStoragePolicy)
+
+        self.action(ICSRFStoragePolicy, register, order=PHASE1_CONFIG)
```

Comment:
I'm pretty sure the `PHASE1_CONFIG` is unnecessary here. It's used in `set_default_csrf_options` because it needs to be configured prior to `add_view` running the `csrf_view` view deriver. I think the storage policy is only used at request-time though so can be left at the default config phase.

If you think it should be `PHASE1_CONFIG` then it should be added to http://docs.pylonsproject.org/projects/pyramid/en/master/narr/extconfig.html#ordering-actions